hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -600,6 +600,24 @@ public void deleteFilesystem(TracingContext tracingContext)
     }
   }
 
+  public OutputStream createPathNonRecursive(final Path path,
+      final FileSystem.Statistics statistics,
+      final boolean overwrite,
+      final FsPermission permission,
+      final FsPermission umask,
+      TracingContext tracingContext)
+      throws IOException {
+    boolean isNamespaceEnabled = getIsNamespaceEnabled(tracingContext);
+    AbfsRestOperation op = getClient().createNonRecursivePath(
+        path.toUri().getPath(), true, overwrite,
+        new Permissions(isNamespaceEnabled, permission,
+            umask),
+        false,
+        null, null, tracingContext, isNamespaceEnabled);
+    return createAbfsOutputStreamInstance(statistics, tracingContext,
```

Comment:
is no check for appendBlob and contextEncryptionAdapter needed for non recursive create ?

Reply:
corrected.